shared/src/main/scala/bloop/io/Environment.scala

```diff
@@ -0,0 +1,117 @@
+package bloop.io
+
+object Environment {
+
+  /**
+   * Returns preferred SHELL environment lineSeparator:
+   *    '\r\n' on Windows, unless SHELL environment is recognized.
+   *    '\n'   everywhere else
+   *
+   * The default ending in Windows is '\r\n', but either of two conditions can
+   * change the value of [[shellLineSeparator]]:
+   *    1. if [[validShell]] is nonEmpty
+   *    2. if system property `"line.separator"` is modified
+   *
+   * @return preferred SHELL lineSeparator, or system property "line.separator"` otherwise.
+   *
+   * @example
+   * {{{
+   *   print(someText + lineSeparator)
+   * }}}
+   */
+  def lineSeparator: String = if (validShell.nonEmpty) "\n" else sysLineSeparator
+
+  /**
+   * Return the canonical lowercased shell name, or empty string.
+   *
+   * `shell` is valid if when lowercased, it has a `validShells` entry as substring.
+   *
+   * @return the matching `validShells` entry, if there is a match; empty string otherwise.
+   *
+   * @example if {{{SHELL=/usr/bin/bash}}}, returns {{{/bin/bash}}}.
+   *
+   */
+  private def validShell: String = validShells.find { _.endsWith(shellLC) } match {
```

Comment:
If `SHELL` is not defined, shellLC will be `""` and this will always return true. Maybe let's just do:
```suggestion
  val lineSeparator: String = Option(System.getenv("SHELL")) match {
    case Some(currentShell) if validShells.exists( sh => currentShell.contains(sh)) => "\n"
    case None => System.getProperty("line.separator", "\n")
  }
```

I am getting a feeling that it's getting a bit overcomplicated with some many different methods, that are basically being used in only one.

Reply:
Although this means SHELL must not have '.exe' or '.Exe' suffix, although it seems like a good simplification.